### ISSUE TITLE
Low level client exception structure should match SimpleJson

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch-net#3656]
With the move to utf8json in #3493, we should ensure that the serialized structured of an exception in the low level client matches that of SimpleJson, the preceding low level client serializer.

This may also be an opportunity to create a better exception structure, to support indexing into Elasticsearch and other logging systems.

**Comment by Mpdreamz** [comment]
we created a custom strategy for exceptions for SimpleJson:

https://github.com/elastic/elasticsearch-net/blob/6.x/src/Elasticsearch.Net/Serialization/ElasticsearchNetJsonStrategy.cs

So we need to match that one not `SimpleJson` default behaviour.

This strategy closely mimics what https://github.com/serilog/serilog-sinks-elasticsearch does when I submitted this PR for its 2.0 https://github.com/serilog/serilog-sinks-elasticsearch/pull/1

We should definitely learn from what has happened on that repos since, lotsa great options and different outputs for different usecases. Even different usecases within just Kibana.

I would love long term for us to maintain a dependency free `Elastic.ExceptionSerializer` that takes a stream to write to. I can see many usecases for this see also e.g: https://github.com/serilog/serilog-sinks-elasticsearch/issues/224. The learnings from that could be applied to ship a dependency free ECS mapping :smile: 


For the initial `7.0.0-alpha` feature parity with `ElasticsearchNetJsonStrategy` would be enough though.

cc @mivano
